### DISCUSSION TITLE
BLE MiFlora Temperature is Celsius but Displayed as Farenheit

### DISCUSSION THREAD

**Discussion by oldjohngalt** [arendst/Tasmota#14644] [Q&A] [answered]
I am using Tasmota 10.1.0 and 2022.01.3 as a BLE to MQTT bridge to read my Mi Flora plant sensor data.

The issue is the data is always presented in Celsius even when I use SetOption8.  The value stays the same but the unit changes.

When I have "SetOption8 0" set so that values are in Celsius then things look normal as below.

![image](https://user-images.githubusercontent.com/28765475/151622976-6303ec20-0197-4088-ae9a-59a031e632f0.png)

Looking at the data in the console it appears the data from the Mi Flora is coming through without the TempUnit parameter that I normally see for other temperature sensors or the ESP32 internal temperature (that I don't care about).
```
22:20:54.260 RSL: SENSOR = {"Time":"2022-01-28T22:20:54","Flora6bxxxxx":{"mac":"c47c8d6xxxxx","Temperature":10.5,"Illuminance":1281,"Moisture":18,"Fertility":198,"RSSI":-91},"Flora6b741e":{"mac":"c47c8d6xxxxx","Temperature":22.3,"Illuminance":0,"Moisture":0,"Fertility":0,"RSSI":-57}}
22:20:56.679 RSL: STATUS10 = {"StatusSNS":{"Time":"2022-01-28T22:20:56","ESP32":{"Temperature":54.4},"TempUnit":"C"}}
```
Therefore when I "SetOption8 1" it displays the values still in Celsius , but it has the degrees F symbol.  Is there a way to change the template so the data from the MiFlora has "TempUnit":"C" in the data?  This should make it so the data will be correctly converted to Fahrenheit.

As a workaround is there another way to convert this value?

Thanks!

**Comment by barbudor** [comment]
Definitively the temperature should be converted in F, and proper unit should be appended.
I'm looking into this

**Reply by oldjohngalt** [reply]
Thank you!!

I saw this other discussion which may or may not be related.  https://github.com/arendst/Tasmota/discussions/14150

**Reply by barbudor** [reply]
Probably not. Publishing of ESP32 temp sensor goes through the normal SENSOR message processing which involve adding the temp unit and properly converting.
BLE sensors are using some shortcuts and not using the same processing. Some code needs to be dupplicated there.

**Comment by barbudor** [comment, marked as answer]
PR done : https://github.com/arendst/Tasmota/pull/14649
If you can't wait for it to be merged, here is a test build
Please upgrade once it has been officially merged (probably tomorrow europe time)
[tasmota32-bluetooth.zip](https://github.com/arendst/Tasmota/files/7962202/tasmota32-bluetooth.zip)

**Reply by barbudor** [reply]
![image](https://user-images.githubusercontent.com/2996042/151630886-1bc756b6-83f6-4e5f-97be-0576a42f0f92.png)
```
 tele/esp32dev3/SENSOR = {"Time":"2022-01-28T23:31:44+01:00","ATC2bdb8e":{"alias":"chbre_lucas","mac":"a4c1382bdb8e","Temperature":66.0,"Humidity":45.9,"DewPoint":44.5,"Btn":1,"Battery":78,"RSSI":-88},"ATC60b9f8":{"alias":"salle_bain","mac":"a4c13860b9f8","Temperature":68.8,"Humidity":59.1,"DewPoint":53.9,"Btn":1,"Battery":65,"RSSI":-65},"TempUnit":"F"} (retained)
 ```

**Comment by oldjohngalt** [comment]
Thank you!  I installed the test firmware and it works great! I will upgrade to the official release once it is merged.

Thanks again!!
